From patchwork Tue Jun  3 19:39:27 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Chris Morgan <macroalpha82@gmail.com>
X-Patchwork-Id: 14106372
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 2EAD3C61CE7
	for <linux-rockchip@archiver.kernel.org>;
 Tue,  3 Jun 2025 19:42:14 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-ID:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=Qb/EyzEbkExOLd0xj+3Nyr6YMkwVTqJZ7K3myMyW2gg=; b=EKuR4TiBVGh30q
	SmyNxdbnIhPTcXWIHGlmVv1AO4qrwhKSOmJGPLKV9c+s/BFH7GBgYbu2+4ds5dwENUPf9842epJCx
	ymSyCAGFWoKx2u67nzgfQF10+Wee82wLD+nETr1QhzpAgcgbkpk89vO5wBVGvuZ6CBeP4hIQBtTBT
	uFj1saq7ccb8zQAc6HVkGplG5jzuASelzTqYkQr+OWNu9g8Fqi5/ZCNDNJR4mCfCgO+qb0ZMbDgqH
	xPkRSWSbuoenDoQUXnVdrOultTuE7/4D/iA7ewFhiHLKAZCAriDfcoymps/E6eflxoEL7NqJIScO+
	/mdg49QPtk03zC7Ktjng==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uMXWN-0000000BjEG-1r0a;
	Tue, 03 Jun 2025 19:42:03 +0000
Received: from mail-oo1-xc36.google.com ([2607:f8b0:4864:20::c36])
	by bombadil.infradead.org with esmtps (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uMXWL-0000000BjCi-0FOp
	for linux-rockchip@lists.infradead.org;
	Tue, 03 Jun 2025 19:42:02 +0000
Received: by mail-oo1-xc36.google.com with SMTP id
 006d021491bc7-60f0a92391bso57989eaf.0
        for <linux-rockchip@lists.infradead.org>;
 Tue, 03 Jun 2025 12:42:00 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1748979720; x=1749584520;
 darn=lists.infradead.org;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:from:to:cc:subject:date
         :message-id:reply-to;
        bh=/L4zSsY66uDYamOrq9+Mw2rOYYJ499EOM1f+pBYr+Z8=;
        b=jyZO8/Y1p6K3IZ25XqIc5UJWLQppIZ6vm5HDkIH07ziBH6Gk/6zvOsin30QZsmHcsL
         3O09x9Tgw9IRisR28kzebA6jVLpzZIKsCx9NWjY8PJIntnoK9kSpKa3H5IU50VaStBea
         mNybf+Yx/kWZW2Zbg6p1sjDY4uA0Bh0HUt0XVq9MUfbTh5fBQlI9yxc+WvPstZ4AsUrv
         WdB701GEgmb1Iatlg5hIvhxIJv/tLTtgSm8K2F+mjXUOX9eZJ6DOc+L5lfFIIbXTjOS2
         iUkB6Pp7EneGrWTzm3UYj7Vd3q8AiSIPVjvF4d6pqhYVnv8374zESnDVNt9YysmMgad9
         t0bg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1748979720; x=1749584520;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-message-state:from:to:cc
         :subject:date:message-id:reply-to;
        bh=/L4zSsY66uDYamOrq9+Mw2rOYYJ499EOM1f+pBYr+Z8=;
        b=bpr/n4UyrO4Ipo1ygbCyAm5ycRyjc4fk4q/KS/k/hd9Z2TbjST23kKapxYVFDCCnT8
         T4NmaRFm/qAIF+sOI7LhgwrMeLg5LIOeAgSmUBiHYfNf4XRaTEhcHPj5tKk2Ogt15Z1e
         BgnfG/b075eKbVeZZVFeEHmD7zt05zZPg9LkDOH1KVmyL0orthch5i9jCv4owvkhMsIF
         wtw/lxhLX9HBdZHXDj9cjLXK5w/AyWVjXNN60pcMiMN3WHi4pFS+aJ0sZ3FxjQJ++4rt
         mIqsopblXWOWU5WDeYDEfAgi9ZeK125ojWTol80lYSGWw0UHMnjl3xQkzmaOPmv4YyV1
         DsUA==
X-Gm-Message-State: AOJu0Yx3iHLaZDEG8ZhqL/Hn2sFj/OLY0KXpyl6u8RUvkmbp+huQ+b+S
	ma9zSqEepqTwo40dWcC0ZJbQDHXIMWyA1hiKJbJ03Q9I69JAVqgmgz+K
X-Gm-Gg: ASbGncvDP17FOvpd4Iy/Y5QGoOhGSQNgg/tFg4jxGuwLqmWEDwJLqooAtQOQopLKEob
	zlAnCvKZHETnwv4b2BYHbQtuUqh93RvfvAmoJrwZTDRZGAphAjhmDDwTNrYUINOjv+NhBfNOzCc
	7NpdFhhcsIsg/14eS01ybSbrMbbagQx5asbCjCERtM+oaqQwX4Mxi8AKaOLKC0/0zTNaLTdG4pI
	13QSVJa2TpM9S/fTadLXe2cBxXj4U0hrMfzzil9/Nb8/ID8sBBD4HTSzUP2zaoShNcZMGLmsx9a
	Z+Br1iulQezwnPuX6PRUmd2OE6dT49cqru1vlWWoCfMSCUN8NW/FybpR2yUppR0gUmDqQwo=
X-Google-Smtp-Source: 
 AGHT+IG3+8X4rYmlVGqs/t382PwbAWeG5RNhDIJA/498vjwFKFaLf3Z5w5vaFUsKAWkzdqYLMZa4cw==
X-Received: by 2002:a4a:ee83:0:b0:60b:b134:a1 with SMTP id
 006d021491bc7-60d53c761b8mr10625534eaf.6.1748979719622;
        Tue, 03 Jun 2025 12:41:59 -0700 (PDT)
Received: from localhost.localdomain
 ([2600:1700:fb0:1bcf:d19a:d10c:4358:1346])
        by smtp.gmail.com with ESMTPSA id
 006d021491bc7-60c14c88f96sm1967615eaf.17.2025.06.03.12.41.58
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Jun 2025 12:41:59 -0700 (PDT)
From: Chris Morgan <macroalpha82@gmail.com>
To: dri-devel@lists.freedesktop.org
Cc: linux-rockchip@lists.infradead.org,
	devicetree@vger.kernel.org,
	sebastian.reichel@collabora.com,
	heiko@sntech.de,
	conor+dt@kernel.org,
	krzk+dt@kernel.org,
	robh@kernel.org,
	tzimmermann@suse.de,
	mripard@kernel.org,
	maarten.lankhorst@linux.intel.com,
	simona@ffwll.ch,
	airlied@gmail.com,
	quic_jesszhan@quicinc.com,
	neil.armstrong@linaro.org,
	javierm@redhat.com,
	megi@xff.cz,
	Chris Morgan <macromorgan@hotmail.com>
Subject: [PATCH 1/4] dt-bindings: vendor-prefixes: Add prefix for Huiling
Date: Tue,  3 Jun 2025 14:39:27 -0500
Message-ID: <20250603193930.323607-2-macroalpha82@gmail.com>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20250603193930.323607-1-macroalpha82@gmail.com>
References: <20250603193930.323607-1-macroalpha82@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20250603_124201_098765_747E0007 
X-CRM114-Status: UNSURE (   8.94  )
X-CRM114-Notice: Please train this message.
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Chris Morgan <macromorgan@hotmail.com>

Shenzhen Huiling Information Technology Co. Ltd. specializes in the
research and manufacturing of display and touch screens for industrial
usage. https://en.szhuiling.com/

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
Acked-by: Krzysztof Kozlowski <krzysztof.kozlowski@linaro.org>
---
 Documentation/devicetree/bindings/vendor-prefixes.yaml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Documentation/devicetree/bindings/vendor-prefixes.yaml b/Documentation/devicetree/bindings/vendor-prefixes.yaml
index c01adbaacbbb..d5fef99a3490 100644
--- a/Documentation/devicetree/bindings/vendor-prefixes.yaml
+++ b/Documentation/devicetree/bindings/vendor-prefixes.yaml
@@ -672,6 +672,8 @@ patternProperties:
     description: Huawei Technologies Co., Ltd.
   "^hugsun,.*":
     description: Shenzhen Hugsun Technology Co. Ltd.
+  "^huiling,.*":
+    description: Shenzhen Huiling Information Technology Co., Ltd.
   "^hwacom,.*":
     description: HwaCom Systems Inc.
   "^hxt,.*":

From patchwork Tue Jun  3 19:39:28 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Chris Morgan <macroalpha82@gmail.com>
X-Patchwork-Id: 14106373
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id E0748C5AD49
	for <linux-rockchip@archiver.kernel.org>;
 Tue,  3 Jun 2025 19:42:13 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-ID:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=dANgNDc8CELIa11HFqZ6daFLVhi1dkhd7bXoy/jGPDs=; b=sjt7hZqwuWEFiN
	0WHl5sJXDrAs/9y6lq5WDlWQYEYV05/4iz6CnQwAfXd5Eim03BQ8KHtgL+ew8Muf8vWCru5f7N4cf
	okzeNAQqI6DIl1oxAx+VqrIg7WyFwjkdd8JcVEf/7RCJy4V9iAF0Oucxrql4WJ8Nu+CwqXf7nbiVM
	qAedPATIXwj09aobavmy05Y75+MdYXktIS8Ww0oJw7O8zuvZLtWA6aojLMxJtaELUkvor6c5hhtxb
	CRVd+iTr3Y4adRYxXb8gRdjbbfU2JO5Z6nDxT1ohbHaZWUtenqOEzhlbfEIBX3hGp17haHsL+rNQr
	9xaJyZy1wSxyjqIiAjyw==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uMXWN-0000000BjEP-3HkH;
	Tue, 03 Jun 2025 19:42:03 +0000
Received: from mail-ot1-x32d.google.com ([2607:f8b0:4864:20::32d])
	by bombadil.infradead.org with esmtps (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uMXWL-0000000BjCw-2fJX
	for linux-rockchip@lists.infradead.org;
	Tue, 03 Jun 2025 19:42:03 +0000
Received: by mail-ot1-x32d.google.com with SMTP id
 46e09a7af769-735a6faec9eso3520685a34.3
        for <linux-rockchip@lists.infradead.org>;
 Tue, 03 Jun 2025 12:42:01 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1748979720; x=1749584520;
 darn=lists.infradead.org;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:from:to:cc:subject:date
         :message-id:reply-to;
        bh=XjzLPAsqasz0SsRAqSr7Vzo2ligN562NWSmZO1lw4nU=;
        b=boXQnXyy5xib20IC6TRvveML3lA4/YINTYEjK3uz0mDGGYvYjpOwey3E7TDZbVYhqt
         /xcASO4oyexcXocZN60xbrG4/9ab4QzAwbqS7otAynGjlzgq/hIfmNslqmPsj9TL1lXf
         E4+sGyYvLF24FX0I8VoLFHldSJZdlChsdV//6lMDuupnfX6M9FYqcbzDfJ0IILIAiYbk
         L/ttzALa3Q5iVfBOxrkVWhcUMsOeEBqvuZKvkTuGfoPIC1D4rOdKVTLqi237DBF/vrij
         AoLpFEhPmxR1oCS+MgQXMXXSpfdl8Bw8xzyVB+9lVeMpsKPFiv0gRKgEbLrZxK8Rfp5N
         jbyw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1748979720; x=1749584520;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-message-state:from:to:cc
         :subject:date:message-id:reply-to;
        bh=XjzLPAsqasz0SsRAqSr7Vzo2ligN562NWSmZO1lw4nU=;
        b=ezpBtJSPUBd/Tf1dIfhl3a/VWmgABdR0nSUSqLcLDuvgAneMI4MakNSgSgj8F0Obsz
         gLUBZJBVAbmwdgtYIfVJPqGj1lxbTuMysUvDXxupl56q4WiFErp6bg8eFjHYEeVunQrx
         sfVAE1aqqeHQ7Qg1NniXQTKwoYctnQvak7x8x+CXEAhX5RxQSEZsMnteDnsxXBWrPYCx
         nZgSfTvqIzdsSz/qKUnz94N+yoQe9h7OETG/cPKCn0WwlyXjl7EF++YAsTZAqo5sGK4o
         Z9WuiyPY7sAiWMA5Zj4USZ4Wivk7G4OrAdFfJjK2VnHkOrMqzrAEL9P2Lx6kBNPGIJKR
         j0cQ==
X-Gm-Message-State: AOJu0YwlPDGngnLVuyTIMgwg6ZiNwIR2Z1h4EiInc6b6q9ttoVUjr7Nc
	TiOCExJirGxerAdxeToe0zrhl7JB2HHVWm5uq9haWamZMLtHFp8iQdfP
X-Gm-Gg: ASbGncvCcs3rZmKNKFlU2Tb3NRZKWiyA5q1lt6TshKLbauLIKNiH1aG1mP760Qu8X1E
	GNPYV84X+YsAQOYIsOCOYPXgEkKQAIDL6DKo4dEhZCoaSJ+QyE534Olh4HvkG0SyhT2mLLDjpG5
	iGCOrle2B6kU2a3VGCj5m+b51c0dldBwsPB3THjf6qMJMWlMxHZ7yl4Hyvykg8Gw6Z1i8EwYN9W
	llTniebEvMsluf7a2WtPFMP1I6j0Otz07LvbT13zO1hkg8IPspNESxRlTPVMzW1OucxA+ysabeP
	OQSomnuoVveoULr179yGzLEvnnORs+CYzcVlh9JUzxw/w5I4hVk6NAVKW1YthJ82EKu6WbkJHre
	u2IlUzA==
X-Google-Smtp-Source: 
 AGHT+IGEdqI2TCTVkK8o7agajIP4kfms0UAr6s+D6iGNsRfQhvMcgU7dK7DNcXKDtlDRpdBJDSTayg==
X-Received: by 2002:a05:6830:d13:b0:72b:9f17:1da0 with SMTP id
 46e09a7af769-73869c39496mr299430a34.7.1748979720415;
        Tue, 03 Jun 2025 12:42:00 -0700 (PDT)
Received: from localhost.localdomain
 ([2600:1700:fb0:1bcf:d19a:d10c:4358:1346])
        by smtp.gmail.com with ESMTPSA id
 006d021491bc7-60c14c88f96sm1967615eaf.17.2025.06.03.12.41.59
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Jun 2025 12:42:00 -0700 (PDT)
From: Chris Morgan <macroalpha82@gmail.com>
To: dri-devel@lists.freedesktop.org
Cc: linux-rockchip@lists.infradead.org,
	devicetree@vger.kernel.org,
	sebastian.reichel@collabora.com,
	heiko@sntech.de,
	conor+dt@kernel.org,
	krzk+dt@kernel.org,
	robh@kernel.org,
	tzimmermann@suse.de,
	mripard@kernel.org,
	maarten.lankhorst@linux.intel.com,
	simona@ffwll.ch,
	airlied@gmail.com,
	quic_jesszhan@quicinc.com,
	neil.armstrong@linaro.org,
	javierm@redhat.com,
	megi@xff.cz,
	Chris Morgan <macromorgan@hotmail.com>
Subject: [PATCH 2/4] dt-bindings: display: himax-hx8394: Add Huiling
 hl055fhav028c
Date: Tue,  3 Jun 2025 14:39:28 -0500
Message-ID: <20250603193930.323607-3-macroalpha82@gmail.com>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20250603193930.323607-1-macroalpha82@gmail.com>
References: <20250603193930.323607-1-macroalpha82@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20250603_124201_675551_72E982D7 
X-CRM114-Status: GOOD (  11.30  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Chris Morgan <macromorgan@hotmail.com>

Add compatible string for the Huiling hl055fhav028c. This panel is
based on the Himax HX8399C display controller which is extremely
similar to the existing HX8394. Add a new constant for
himax,hx8399c for this new display controller as well.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
Acked-by: Krzysztof Kozlowski <krzysztof.kozlowski@linaro.org>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
---
 .../bindings/display/panel/himax,hx8394.yaml    | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/Documentation/devicetree/bindings/display/panel/himax,hx8394.yaml b/Documentation/devicetree/bindings/display/panel/himax,hx8394.yaml
index 75ccabff308b..5725a587e35c 100644
--- a/Documentation/devicetree/bindings/display/panel/himax,hx8394.yaml
+++ b/Documentation/devicetree/bindings/display/panel/himax,hx8394.yaml
@@ -17,12 +17,17 @@ description:
 
 properties:
   compatible:
-    items:
-      - enum:
-          - hannstar,hsd060bhw4
-          - microchip,ac40t08a-mipi-panel
-          - powkiddy,x55-panel
-      - const: himax,hx8394
+    oneOf:
+      - items:
+          - enum:
+              - hannstar,hsd060bhw4
+              - microchip,ac40t08a-mipi-panel
+              - powkiddy,x55-panel
+          - const: himax,hx8394
+      - items:
+          - enum:
+              - huiling,hl055fhav028c
+          - const: himax,hx8399c
 
   reg:
     maxItems: 1

From patchwork Tue Jun  3 19:39:29 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Chris Morgan <macroalpha82@gmail.com>
X-Patchwork-Id: 14106371
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id C2FF3C61CE5
	for <linux-rockchip@archiver.kernel.org>;
 Tue,  3 Jun 2025 19:42:13 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-ID:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=oN9IIPWKPCMmmckBJNE1j3mRhRc81hNO7QB8J124lnk=; b=erUeszDxUa6s83
	Brrlq3ORB+fn6wduRdtvZplQhO3R4VlaUd9msQAnNCZQACfPmeKR0dxZhK6Li4LPsoBqcVoBMx88Z
	lUlK3UqUh1T3aDoiZFRePbd+IvZnSn3Do9wXnVbP5HbDRr0DMAySZlMPdusJdZOxp9vhxnJ8Onjiv
	EB+ruojIuOgH8hP+mx1IvGmeIsxs/WvvuzcXEd/xJ0+C4hyVe3UgONMtpwlZYgvA6b061t4btp6uM
	EjDXdB6wBgiCu4JGtnHbitdOeVO0maQL60fGRFaUyHy2nL2tjTQRDNoTDYEgPTk2InboAOAxFy+FV
	NMThTTnB0E2cQzUp9axg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uMXWP-0000000BjEb-0U6f;
	Tue, 03 Jun 2025 19:42:05 +0000
Received: from mail-oo1-xc35.google.com ([2607:f8b0:4864:20::c35])
	by bombadil.infradead.org with esmtps (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uMXWM-0000000BjDN-1bRy
	for linux-rockchip@lists.infradead.org;
	Tue, 03 Jun 2025 19:42:03 +0000
Received: by mail-oo1-xc35.google.com with SMTP id
 006d021491bc7-60bef8e7bb3so2720116eaf.2
        for <linux-rockchip@lists.infradead.org>;
 Tue, 03 Jun 2025 12:42:01 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1748979721; x=1749584521;
 darn=lists.infradead.org;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:from:to:cc:subject:date
         :message-id:reply-to;
        bh=TIVqz2dRGfkeXaVkRfMlKjXlMd9pJR9pSIQQIKcgCz8=;
        b=JXH/m7oPKBsicm49bSJkdYnPcm819vNojYwBmYiIODqtueOqTUmZVDu2VfRprebO6B
         783N12xf8SzzIdZO5VVk3PysjzarzPd0srZdKKUJS5hAgFWwGUo0SKT1cvh1IszxCErr
         ZWzCogYllN4WtPO/a9ryL5GQO8EjPtjBZkd5WUUXP9nlSNU4/kmU6iLGsTWXmv65TwEE
         3kMvwz/Xk+LGbTgOl750qMFP2vxcVk2X7vTCnvCDtEasQJhgNNnN1qBv61nT23Vjo5qY
         o2xPhzsOxkiu1CHofbTIt13pPJZakZgMSnWUIMp4saSB1UaPkN0lLuSfetoCW12Jv30L
         T7Sw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1748979721; x=1749584521;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-message-state:from:to:cc
         :subject:date:message-id:reply-to;
        bh=TIVqz2dRGfkeXaVkRfMlKjXlMd9pJR9pSIQQIKcgCz8=;
        b=F8DsAN1MR29tmrlpWY8pT9mjqpftEw2iJARuPkhouhQkdiVenSXpDsx5r4/rtaLg8y
         oSR0+POS+rFCki7i7oyf2DC1wo5F821o8vMgUasw2tVhxMg65J6XGo5WcQuu2s0KGe+K
         5mWSM6Nb2li+q/xYLd2bRkEjwJQ2CBp+huxw5C3PKYL4e2hSevlisQ3SJlkf+/557EZi
         ULPEUQTp60kKF90WI+klnTUNxq5HNI0ogP4IoJsDSXNh9oiKSOADlxPMY972ZS+VBzEz
         bMPLLq8QeAw8yMXdWuJS/6uvGI6/iCPGfDQJpMs4NfNGjsFwL03UbkQC06xcZ0+GDzyp
         ohrQ==
X-Gm-Message-State: AOJu0Yw097OeNujURXT+PxlchwaofO1OTz4V3Ux6GBA6LwTg7HPmGioV
	ewLjULgBP43h+QxE5Ef5ePANxd+IgovNmBlAIsFUAh8IeN9Lb0o130qU
X-Gm-Gg: ASbGncvqYfU1cXNZvxy8tDzu4Hn9NcqxsSWoDwRr2Q7fmXu8gXrYDLQtAfOty7ZXJ0H
	x/assvwvXPh3nR5hIdRlS6Y+CQvYkDEXk0reimVp2IIzSmBXYbfGm/cQecpIvdlV1d4dtwIggLM
	+O1ANSGjhR8TeSpJLtY94RnamfN2TzNJy0v/FyJNf4Go0/uVaatzYmApvzrUEC214L6/QfkVrMK
	soWxw8oMToYRqscd5JyknnrtpSvlyUKCK4cBELmOmKBxIL3Ttz0ZioRSAtIxd5bZrIQeXf025NV
	U+cQCbB9AYSqIZh9tzuTO1eE1FQpEYHHbIzKJKmwfhXXNuh/toVXFiGPq7JKTdhSLP2p1EQ=
X-Google-Smtp-Source: 
 AGHT+IEPHZuFkTYXQCTkJo5Ym0ylfV4pxcvfbfHQSSmEIozDMoZD4JWFzvzgmrQ3ty+jtqiGY0g7Aw==
X-Received: by 2002:a05:6820:2901:b0:60e:e75c:5ef2 with SMTP id
 006d021491bc7-60ee75c6b78mr6812635eaf.7.1748979721192;
        Tue, 03 Jun 2025 12:42:01 -0700 (PDT)
Received: from localhost.localdomain
 ([2600:1700:fb0:1bcf:d19a:d10c:4358:1346])
        by smtp.gmail.com with ESMTPSA id
 006d021491bc7-60c14c88f96sm1967615eaf.17.2025.06.03.12.42.00
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Jun 2025 12:42:00 -0700 (PDT)
From: Chris Morgan <macroalpha82@gmail.com>
To: dri-devel@lists.freedesktop.org
Cc: linux-rockchip@lists.infradead.org,
	devicetree@vger.kernel.org,
	sebastian.reichel@collabora.com,
	heiko@sntech.de,
	conor+dt@kernel.org,
	krzk+dt@kernel.org,
	robh@kernel.org,
	tzimmermann@suse.de,
	mripard@kernel.org,
	maarten.lankhorst@linux.intel.com,
	simona@ffwll.ch,
	airlied@gmail.com,
	quic_jesszhan@quicinc.com,
	neil.armstrong@linaro.org,
	javierm@redhat.com,
	megi@xff.cz,
	Chris Morgan <macromorgan@hotmail.com>
Subject: [PATCH 3/4] drm/panel: himax-hx8394: Add Support for Huiling
 hl055fhav028c
Date: Tue,  3 Jun 2025 14:39:29 -0500
Message-ID: <20250603193930.323607-4-macroalpha82@gmail.com>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20250603193930.323607-1-macroalpha82@gmail.com>
References: <20250603193930.323607-1-macroalpha82@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20250603_124202_444995_07CE939C 
X-CRM114-Status: GOOD (  14.59  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Chris Morgan <macromorgan@hotmail.com>

Add support for the Huiling hl055fhav028c panel as used on the
Gameforce Ace handheld gaming console. This panel uses a Himax HX8399C
display controller and requires a sparsely documented vendor provided
init sequence. The display resolution is 1080x1920 and is 70mm by 127mm
as stated in the manufacturer's documentation.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
Reviewed-by: Jessica Zhang <quic_jesszhan@quicinc.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
---
 drivers/gpu/drm/panel/panel-himax-hx8394.c | 142 +++++++++++++++++++++
 1 file changed, 142 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-himax-hx8394.c b/drivers/gpu/drm/panel/panel-himax-hx8394.c
index ff994bf0e3cc..16e450b156b7 100644
--- a/drivers/gpu/drm/panel/panel-himax-hx8394.c
+++ b/drivers/gpu/drm/panel/panel-himax-hx8394.c
@@ -477,6 +477,147 @@ static const struct hx8394_panel_desc mchp_ac40t08a_desc = {
 	.init_sequence = mchp_ac40t08a_init_sequence,
 };
 
+/*
+ * HL055FHAV028C is based on Himax HX8399, so datasheet pages are
+ * slightly different than HX8394 based panels.
+ */
+static void hl055fhav028c_init_sequence(struct mipi_dsi_multi_context *dsi_ctx)
+{
+	/* 6.3.6 SETEXTC: Set extension command (B9h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETEXTC,
+				     0xff, 0x83, 0x99);
+
+	/* 6.3.17 SETOFFSET: Set offset voltage (D2h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETOFFSET,
+				     0x77);
+
+	/* 6.3.1 SETPOWER: Set power (B1h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETPOWER,
+				     0x02, 0x04, 0x74, 0x94, 0x01, 0x32,
+				     0x33, 0x11, 0x11, 0xab, 0x4d, 0x56,
+				     0x73, 0x02, 0x02);
+
+	/* 6.3.2 SETDISP: Set display related register (B2h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETDISP,
+				     0x00, 0x80, 0x80, 0xae, 0x05, 0x07,
+				     0x5a, 0x11, 0x00, 0x00, 0x10, 0x1e,
+				     0x70, 0x03, 0xd4);
+
+	/* 6.3.3 SETCYC: Set display waveform cycles (B4h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETCYC,
+				     0x00, 0xff, 0x02, 0xc0, 0x02, 0xc0,
+				     0x00, 0x00, 0x08, 0x00, 0x04, 0x06,
+				     0x00, 0x32, 0x04, 0x0a, 0x08, 0x21,
+				     0x03, 0x01, 0x00, 0x0f, 0xb8, 0x8b,
+				     0x02, 0xc0, 0x02, 0xc0, 0x00, 0x00,
+				     0x08, 0x00, 0x04, 0x06, 0x00, 0x32,
+				     0x04, 0x0a, 0x08, 0x01, 0x00, 0x0f,
+				     0xb8, 0x01);
+
+	/* 6.3.18 SETGIP0: Set GIP Option0 (D3h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETGIP0,
+				     0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+				     0x06, 0x00, 0x00, 0x10, 0x04, 0x00,
+				     0x04, 0x00, 0x00, 0x00, 0x00, 0x00,
+				     0x00, 0x00, 0x00, 0x00, 0x00, 0x01,
+				     0x00, 0x05, 0x05, 0x07, 0x00, 0x00,
+				     0x00, 0x05, 0x40);
+
+	/* 6.3.19 Set GIP Option1 (D5h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETGIP1,
+				     0x18, 0x18, 0x19, 0x19, 0x18, 0x18,
+				     0x21, 0x20, 0x01, 0x00, 0x07, 0x06,
+				     0x05, 0x04, 0x03, 0x02, 0x18, 0x18,
+				     0x18, 0x18, 0x18, 0x18, 0x2f, 0x2f,
+				     0x30, 0x30, 0x31, 0x31, 0x18, 0x18,
+				     0x18, 0x18);
+
+	/* 6.3.20 Set GIP Option2 (D6h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETGIP2,
+				     0x18, 0x18, 0x19, 0x19, 0x40, 0x40,
+				     0x20, 0x21, 0x02, 0x03, 0x04, 0x05,
+				     0x06, 0x07, 0x00, 0x01, 0x40, 0x40,
+				     0x40, 0x40, 0x40, 0x40, 0x2f, 0x2f,
+				     0x30, 0x30, 0x31, 0x31, 0x40, 0x40,
+				     0x40, 0x40);
+
+	/* 6.3.21 Set GIP Option3 (D8h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_UNKNOWN4,
+				     0xa2, 0xaa, 0x02, 0xa0, 0xa2, 0xa8,
+				     0x02, 0xa0, 0xb0, 0x00, 0x00, 0x00,
+				     0xb0, 0x00, 0x00, 0x00);
+
+	/* 6.3.9 Set register bank (BDh) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETREGBANK,
+				     0x01);
+
+	/* 6.3.21 Set GIP Option3 (D8h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_UNKNOWN4,
+				     0xb0, 0x00, 0x00, 0x00, 0xb0, 0x00,
+				     0x00, 0x00, 0xe2, 0xaa, 0x03, 0xf0,
+				     0xe2, 0xaa, 0x03, 0xf0);
+
+	/* 6.3.9 Set register bank (BDh) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETREGBANK,
+				     0x02);
+
+	/* 6.3.21 Set GIP Option3 (D8h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_UNKNOWN4,
+				     0xe2, 0xaa, 0x03, 0xf0, 0xe2, 0xaa,
+				     0x03, 0xf0);
+
+	/* 6.3.9 Set register bank (BDh) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETREGBANK,
+				     0x00);
+
+	/* 6.3.4 SETVCOM: Set VCOM voltage (B6h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETVCOM,
+				     0x7a, 0x7a);
+
+	/* 6.3.26 SETGAMMA: Set gamma curve related setting (E0h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETGAMMA,
+				     0x00, 0x18, 0x27, 0x24, 0x5a, 0x68,
+				     0x79, 0x78, 0x81, 0x8a, 0x92, 0x99,
+				     0x9e, 0xa7, 0xaf, 0xb4, 0xb9, 0xc3,
+				     0xc7, 0xd1, 0xc6, 0xd4, 0xd5, 0x6c,
+				     0x67, 0x71, 0x77, 0x00, 0x00, 0x18,
+				     0x27, 0x24, 0x5a, 0x68, 0x79, 0x78,
+				     0x81, 0x8a, 0x92, 0x99, 0x9e, 0xa7,
+				     0xaf, 0xb4, 0xb9, 0xc3, 0xc7, 0xd1,
+				     0xc6, 0xd4, 0xd5, 0x6c, 0x67, 0x77);
+
+	/* Unknown command, not listed in the HX8399-C datasheet (C6h) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_UNKNOWN2,
+				     0xff, 0xf9);
+
+	/* 6.3.16 SETPANEL (CCh) */
+	mipi_dsi_dcs_write_seq_multi(dsi_ctx, HX8394_CMD_SETPANEL,
+				     0x08);
+}
+
+static const struct drm_display_mode hl055fhav028c_mode = {
+	.hdisplay	= 1080,
+	.hsync_start	= 1080 + 32,
+	.hsync_end	= 1080 + 32 + 8,
+	.htotal		= 1080 + 32 + 8 + 32,
+	.vdisplay	= 1920,
+	.vsync_start	= 1920 + 16,
+	.vsync_end	= 1920 + 16 + 2,
+	.vtotal		= 1920 + 16 + 2 + 14,
+	.clock		= 134920,
+	.flags		= DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC,
+	.width_mm	= 70,
+	.height_mm	= 127,
+};
+
+static const struct hx8394_panel_desc hl055fhav028c_desc = {
+	.mode = &hl055fhav028c_mode,
+	.lanes = 4,
+	.mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST,
+	.format = MIPI_DSI_FMT_RGB888,
+	.init_sequence = hl055fhav028c_init_sequence,
+};
+
 static int hx8394_enable(struct drm_panel *panel)
 {
 	struct hx8394 *ctx = panel_to_hx8394(panel);
@@ -683,6 +824,7 @@ static void hx8394_remove(struct mipi_dsi_device *dsi)
 
 static const struct of_device_id hx8394_of_match[] = {
 	{ .compatible = "hannstar,hsd060bhw4", .data = &hsd060bhw4_desc },
+	{ .compatible = "huiling,hl055fhav028c", .data = &hl055fhav028c_desc },
 	{ .compatible = "powkiddy,x55-panel", .data = &powkiddy_x55_desc },
 	{ .compatible = "microchip,ac40t08a-mipi-panel", .data = &mchp_ac40t08a_desc },
 	{ /* sentinel */ }

From patchwork Tue Jun  3 19:39:30 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Chris Morgan <macroalpha82@gmail.com>
X-Patchwork-Id: 14106370
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 52306C5B552
	for <linux-rockchip@archiver.kernel.org>;
 Tue,  3 Jun 2025 19:42:13 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-ID:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=hCmtDxrSWn+2yQ3NAtmrrj9vdW0secPSu78WImJiuGc=; b=Aka0KtmQI0P1BH
	i1Yx7GZqX6K13KycN2KUIuIkK2S0YJ4xC1tlAh2qscpLmWvD82KwA/QjZIojcvbD+1tG8SPn2aIqL
	sQk+bVUQgDQDBCShvsz5GujI//uLiMGjAM0Pba1yLWGNKPoJ826jNB3m4FqelH7LKxhKWoIXlWTRo
	uwgzMpWxJlJZV3Q3axZcrsq5TDy7dPiynvEIV5PBQqjd7xAIXmiUTDgTwMeRPdAwLsvUWUy+HnZIa
	ZpoHqo0utS2N3TCuTP7Tm9YAxlpHhA5E8rjN75uJ4sRprVFldcNuvly4ai86UqqbrwOwUu7BrsOnh
	I1/BkppCL3iMVvl3gxxA==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uMXWP-0000000BjF7-21Q9;
	Tue, 03 Jun 2025 19:42:05 +0000
Received: from mail-oo1-xc2e.google.com ([2607:f8b0:4864:20::c2e])
	by bombadil.infradead.org with esmtps (Exim 4.98.2 #2 (Red Hat Linux))
	id 1uMXWN-0000000BjDq-0VrR
	for linux-rockchip@lists.infradead.org;
	Tue, 03 Jun 2025 19:42:04 +0000
Received: by mail-oo1-xc2e.google.com with SMTP id
 006d021491bc7-606668f8d51so143711eaf.0
        for <linux-rockchip@lists.infradead.org>;
 Tue, 03 Jun 2025 12:42:02 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1748979722; x=1749584522;
 darn=lists.infradead.org;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:from:to:cc:subject:date
         :message-id:reply-to;
        bh=FgLtrMgHhQ/oGFMvm8sSOBZ5dDdVEtN1pzHrZf4N+fU=;
        b=JKPy7lpqE9aSkYBNDtip80ijXisoDBLp4oiQzMstokRl/J/FvGwX7XsYVzaQF+UqX8
         gATPfDaXn/6jmmAa56uY2PZBTzEmDKkurM+2zeu0lUA7Az5HNKRCgM9+F3WEWKNI9ORq
         m6esRXPv7knzVSCM9okFkzdKH4BMMA68ozL0WJrpVh/YkBzzfSElk2psywj96meklqtM
         C9kUOG/hghNrJ9vT10lbNwBWiupCGwj/iH1byZEqVT27nmLjMKnN19+xHGSewRBNbzEg
         Xc6e4swYBaBmVTqmEB6p74SgMyr5JxF6TRqky2pxG1i64OKnd+8BF/lFZ3fB/nTv/cET
         ptqA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1748979722; x=1749584522;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-message-state:from:to:cc
         :subject:date:message-id:reply-to;
        bh=FgLtrMgHhQ/oGFMvm8sSOBZ5dDdVEtN1pzHrZf4N+fU=;
        b=wh6cxAoDwVpaatNE0a9Tn/7VlYKeiZ3bcEBe2DKvyQg3krNfG5BES2VTLCi+GP6bMA
         wiQage10/pjSHqls+XHs49pv+ESZtz0bnG6E+3NmfBGdeLOPkzwFtL9CxoVmraB9s33B
         r6+NdFE6Q82PL7mhOFETnD9B5whx4wQ+vrhoi1og+Nep8si1pyFdin9zFrTCoLtTIZIV
         RkCoxzuGwxWEPnNz7QQS6mqPcJouxIWdZnRDUZenKtsX3HB8IfpTPtgXaFh6Rfi9mf+e
         qztxZz6O4gC6JR1bWaPN/5SEmUZG0t7NTjaRlBOHMijV5r1up1gF5/NySmfQFUzHZuQo
         hMdQ==
X-Gm-Message-State: AOJu0Yw+eXiVTEannUBWBEr4V6x0BSL8PJHZuQBQoCywCwhPK9h6Frmm
	5CLwLW2lk1zu7F8bdG1GM6s59VKC0XxbsIqVlAdnvnEn0JD7j5Ylpq0x
X-Gm-Gg: ASbGncvbB+O4ireBm/qPVHVtrT7/zUSR4iakzmQOwrSAOUoUI8uk1XYxlhXj2+HW6mn
	IXybIWhjQyQ/m3Q86dDWQUX0TRpz4+jwhYH/ZYKY9SGyzYVH9OdKlJQ3CuL2buEFh0A3DHP5vt1
	qK3Sw89yiHY7DywaRgjA+kYVqbrCLU+n/9+6sE/pjI5+YfFxJnYeNHpVwlSzYMPGDsqKY6w16wJ
	ZlQquEPm2fQaXgFj5s509ZvS+YplE+sa5MyTgI7yD8qT6hhhnNaZnu7v78j5gA06uGAcLjs60jM
	OmxL5NMtP4uS40iifsMxdSAGqLybeMRLJPtGKhfCR+6IwLKtWgJ+Ajh+IVboFedWWrXJ43I=
X-Google-Smtp-Source: 
 AGHT+IE9tiwKq/R/e3U5EawJb79KooQ4IxNgVCBv1cslv8xmu0yJpXxxMeDxU3qVBvLMhI+JZsrssA==
X-Received: by 2002:a4a:dc48:0:b0:608:3493:b807 with SMTP id
 006d021491bc7-60efdd1f98dmr1851832eaf.2.1748979722009;
        Tue, 03 Jun 2025 12:42:02 -0700 (PDT)
Received: from localhost.localdomain
 ([2600:1700:fb0:1bcf:d19a:d10c:4358:1346])
        by smtp.gmail.com with ESMTPSA id
 006d021491bc7-60c14c88f96sm1967615eaf.17.2025.06.03.12.42.01
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 03 Jun 2025 12:42:01 -0700 (PDT)
From: Chris Morgan <macroalpha82@gmail.com>
To: dri-devel@lists.freedesktop.org
Cc: linux-rockchip@lists.infradead.org,
	devicetree@vger.kernel.org,
	sebastian.reichel@collabora.com,
	heiko@sntech.de,
	conor+dt@kernel.org,
	krzk+dt@kernel.org,
	robh@kernel.org,
	tzimmermann@suse.de,
	mripard@kernel.org,
	maarten.lankhorst@linux.intel.com,
	simona@ffwll.ch,
	airlied@gmail.com,
	quic_jesszhan@quicinc.com,
	neil.armstrong@linaro.org,
	javierm@redhat.com,
	megi@xff.cz,
	Chris Morgan <macromorgan@hotmail.com>
Subject: [PATCH 4/4] arm64: dts: rockchip: Add DSI panel support for
 gameforce-ace
Date: Tue,  3 Jun 2025 14:39:30 -0500
Message-ID: <20250603193930.323607-5-macroalpha82@gmail.com>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20250603193930.323607-1-macroalpha82@gmail.com>
References: <20250603193930.323607-1-macroalpha82@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20250603_124203_170645_B81FA0ED 
X-CRM114-Status: GOOD (  11.62  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Chris Morgan <macromorgan@hotmail.com>

Enable the DSI controller, DSI DCPHY, and Huiling hl055fhav028c
1080x1920 panel for the Gameforce Ace.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
---
 .../dts/rockchip/rk3588s-gameforce-ace.dts    | 66 +++++++++++++++++++
 1 file changed, 66 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-gameforce-ace.dts b/arch/arm64/boot/dts/rockchip/rk3588s-gameforce-ace.dts
index 873a2bd6a6de..bb7c1b732cc2 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-gameforce-ace.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-gameforce-ace.dts
@@ -7,6 +7,7 @@
 #include <dt-bindings/leds/common.h>
 #include <dt-bindings/pinctrl/rockchip.h>
 #include <dt-bindings/pwm/pwm.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include <dt-bindings/thermal/thermal.h>
 #include <dt-bindings/usb/pd.h>
 #include "rk3588s.dtsi"
@@ -456,6 +457,42 @@ &cpu_b3 {
 	cpu-supply = <&vdd_cpu_big1_s0>;
 };
 
+&dsi0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	panel@0 {
+		compatible = "huiling,hl055fhav028c", "himax,hx8399c";
+		reg = <0>;
+		backlight = <&backlight>;
+		iovcc-supply = <&vcc3v3_lcd0_n>;
+		pinctrl-0 = <&lcd_rst>;
+		pinctrl-names = "default";
+		reset-gpios = <&gpio1 RK_PD5 GPIO_ACTIVE_LOW>;
+		rotation = <90>;
+		vcc-supply = <&vcc3v3_lcd0_n>;
+
+		port {
+			mipi_panel_in: endpoint {
+				remote-endpoint = <&dsi0_out_panel>;
+			};
+		};
+	};
+};
+
+&dsi0_in {
+	dsi0_in_vp3: endpoint {
+		remote-endpoint = <&vp3_out_dsi0>;
+	};
+};
+
+&dsi0_out {
+	dsi0_out_panel: endpoint {
+		remote-endpoint = <&mipi_panel_in>;
+	};
+};
+
 &gpu {
 	mali-supply = <&vdd_gpu_s0>;
 	status = "okay";
@@ -633,6 +670,10 @@ &i2s0_sdi0
 	status = "okay";
 };
 
+&mipidcphy0 {
+	status = "okay";
+};
+
 &package_thermal {
 	polling-delay = <1000>;
 
@@ -769,6 +810,13 @@ lcd_bl_en: lcd-bl-en {
 		};
 	};
 
+	lcd_rst {
+		lcd_rst: lcd-rst {
+			rockchip,pins =
+				<1 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
 	pcie-pins {
 		pcie_rst: pcie-rst {
 			rockchip,pins =
@@ -1239,3 +1287,21 @@ bluetooth {
 		shutdown-gpios = <&gpio3 RK_PB7 GPIO_ACTIVE_HIGH>;
 	};
 };
+
+&vop {
+	status = "okay";
+};
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vp3 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	vp3_out_dsi0: endpoint@ROCKCHIP_VOP2_EP_MIPI0 {
+		reg = <ROCKCHIP_VOP2_EP_MIPI0>;
+		remote-endpoint = <&dsi0_in_vp3>;
+	};
+};
